pkg/discovery/util/node_util.go

```diff
@@ -185,6 +192,21 @@ func DetectNodeRoles(node *api.Node) sets.String {
 	return allRoles
 }
 
+func DetectNodePools(node *api.Node) sets.String {
+	allPools := sets.NewString()
+	for k, v := range node.Labels {
+		if isValidNodePoolLabel(k) && v != "" {
+			allPools.Insert(v)
+		}
+	}
+	return allPools
+}
+
+func isValidNodePoolLabel(label string) bool {
+	return label == NodePoolAKS || label == NodePoolGKE ||
+		strings.Contains(label, NodePoolEKSIdentifier)
```

Comment:
Query: Why are we not using the exact key string that EKS labels nodepools as.
You seem to have listed that in comments above `alpha.eksctl.io/nodegroup-name or eks.amazonaws.com/nodegroup`

Reply:
We did not clear specification about the node labels for node groups. We came across these node pools while working different clusters. For generality, it is better to look for the nodegroup key word in the label name.